packages/flutter/lib/src/material/chip.dart

```diff
@@ -1342,18 +1342,22 @@ class _RawChipState extends State<RawChip> with TickerProviderStateMixin<RawChip
     if (!hasDeleteButton) {
       return null;
     }
-    return _wrapWithTooltip(
-      new InkResponse(
-        onTap: widget.isEnabled ? widget.onDeleted : null,
-        child: new IconTheme(
-          data: theme.iconTheme.copyWith(
-            color: (widget.deleteIconColor ?? chipTheme.deleteIconColor) ?? theme.iconTheme.color,
+    final IconThemeData iconThemeData = theme.iconTheme.copyWith(
+      color: widget.deleteIconColor ?? chipTheme.deleteIconColor,
+    );
+    return KeyedSubtree.wrap(
+      _wrapWithTooltip(
+        new InkResponse(
+          onTap: widget.isEnabled ? widget.onDeleted : null,
+          child: new IconTheme(
+            data: iconThemeData,
+            child: widget.deleteIcon,
           ),
-          child: widget.deleteIcon,
         ),
+        widget.deleteButtonTooltipMessage ?? MaterialLocalizations.of(context)?.deleteButtonTooltip,
+        widget.onDeleted,
       ),
-      widget.deleteButtonTooltipMessage ?? MaterialLocalizations.of(context)?.deleteButtonTooltip,
-      widget.onDeleted,
+      hashValues(iconThemeData, widget.deleteIcon),
```

Comment:
hashValues isn't unique. This will cause flakes.

Reply:
As far as I can tell you don't really need to have a key here. What matters is whether there is a button or not; not what colour it is (e.g. imagine how this will work when the theme animates).

Reply:
maybe key it based on a new kind of key, say `WidgetIdentityKey`, that takes a _Widget and delegates to canUpdate to compare instances...

Reply:
(and just pass deleteIcon)